contracts/contracts/strategies/sonic/SonicSwapXAMOStrategy.sol

```diff
@@ -0,0 +1,710 @@
+        gauge = _gauge;
+
+        // This is an implementation contract. The governor is set in the proxy contract.
+        _setGovernor(address(0));
+    }
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as SwapX strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of SWPx token
+     */
+    function initialize(address[] calldata _rewardTokenAddresses)
+        external
+        onlyGovernor
+        initializer
+    {
+        address[] memory pTokens = new address[](1);
+        pTokens[0] = pool;
+
+        address[] memory _assets = new address[](1);
+        _assets[0] = ws;
+
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            pTokens
+        );
+
+        _approveBase();
+    }
+
+    /***************************************
+                    Deposit
+    ****************************************/
+
+    /**
+     * @notice Deposit Wrapped S (wS) into the SwapX pool
+     * @param _asset Address of Wrapped S (wS) contract.
+     * @param _amount Amount of Wrapped S (wS) to deposit.
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+        skimPool
+    {
+        _deposit(_asset, _amount);
+    }
+
+    function _deposit(address _wS, uint256 _wsAmount) internal {
+        require(_wsAmount > 0, "Must deposit something");
+        require(_wS == ws, "Unsupported asset");
+
+        // Calculate the required amount of OS to mint based on the wS amount.
+        uint256 osDepositAmount = _calcTokensToMint(_wsAmount);
+
+        // Mint the required OS tokens to this strategy
+        IVault(vaultAddress).mintForStrategy(osDepositAmount);
+
+        // Add wS and OS liquidity to the pool and stake in gauge
+        _depositToPoolAndGauge(_wsAmount, osDepositAmount);
+
+        // Ensure solvency of the vault
+        _solvencyAssert();
+
+        // Emit event for the deposited wS tokens
+        emit Deposit(_wS, pool, _wsAmount);
+        // Emit event for the minted OS tokens
+        emit Deposit(os, pool, osDepositAmount);
+    }
+
+    /**
+     * @notice Deposit the strategy's entire balance of Wrapped S (wS) into the pool
+     */
+    function depositAll() external override onlyVault nonReentrant skimPool {
+        uint256 balance = IERC20(ws).balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(ws, balance);
+        }
+    }
+
+    /***************************************
+                    Withdraw
+    ****************************************/
+
+    /**
+     * @notice Withdraw wS and OS from the SwapX pool, burn the OS,
+     * and transfer the wS to the recipient.
+     * @param _recipient Address to receive withdrawn asset which is normally the Vault.
```

Comment:
nit: the vault should be the only valid recipient

Reply:
I've updated the Natspec